### PR TITLE
extract integrations into their own scripts relying on json output

### LINKED ISSUE #385: Provide json as base output
FLOSS should use a standardized json base output to enhance interoperability with external tools and ease the various output modes (cmd line, IDA script, etc.).

related: #188

### LINKED ISSUE #390: extract x64dbg integration into its own script
move this out of FLOSS core and into a supporting script

use #385 json document as the source

### LINKED ISSUE #391: extract idapython integration into its own script
move this out of FLOSS core and into a supporting script

use #385 json document as the source

### LINKED ISSUE #392: extract r2 integration into its own script
move this out of FLOSS core and into a supporting script

use #385 json document as the source

### LINKED ISSUE #393: extract binja integration into its own script
move this out of FLOSS core and into a supporting script

use #385 json document as the source

### LINKED ISSUE #394: extract ghidra integration into its own script
move this out of FLOSS core and into a supporting script

use #385 json document as the source

### PR DESCRIPTION
**Pull request by williballenthin**
- closes #385 (json output)
  - moves integrations into their own scripts
    - closes #390
    - closes #391
    - closes #392
    - closes #393
    - closes #394

current cli help & arguments:
image

### CODE REVIEW

File: floss/main.py [hunk 1]
```diff
@@ -1062,153 +599,149 @@ def main(argv=None):
...
-    min_length = parse_min_length_option(options.min_length)
+    set_log_config(args)
 
     # expert profile settings
-    if options.expert:
-        options.save_workspace = True
-        options.group_functions = True
-        options.quiet = False
+    # TODO: removeme
+    args.expert = args.x
```
Comment (mr-tz) [inline floss/main.py:609]: yes!

**Review by mr-tz — COMMENTED** [review]
great effort already!

File: floss/main.py [hunk 2]
```diff
@@ -1062,153 +599,149 @@ def main(argv=None):
...
+    args.no_static_strings = args.no_static_strings if hasattr(args, "no_static_strings") else False
+    args.no_decoded_strings = args.no_decoded_strings if hasattr(args, "no_decoded_strings") else False
+    args.no_stack_strings = args.no_stack_strings if hasattr(args, "no_stack_strings") else False
+    args.is_shellcode = args.is_shellcode if hasattr(args, "is_shellcode") else False
+    args.shellcode_entry_point = args.shellcode_entry_point if hasattr(args, "shellcode_entry_point") else None
+    args.shellcode_base = args.shellcode_base if hasattr(args, "shellcode_base") else None
+
+    sample_file_path = validate_sample_path(parser, args)
 
     result_document = ResultDocument(
```
Comment (williballenthin) [inline floss/main.py:634]: maybe we should provide an easier constructor here.

File: floss/main.py [hunk 3]
```diff
@@ -1062,153 +599,152 @@ def main(argv=None):
...
+    )
+    args.max_address_revisits = (
+        args.max_address_revisits if hasattr(args, "max_address_revisits") else DEFAULT_MAX_ADDRESS_REVISITS
+    )
+    args.no_static_strings = args.no_static_strings if hasattr(args, "no_static_strings") else False
+    args.no_decoded_strings = args.no_decoded_strings if hasattr(args, "no_decoded_strings") else False
+    args.no_stack_strings = args.no_stack_strings if hasattr(args, "no_stack_strings") else False
+    args.is_shellcode = args.is_shellcode if hasattr(args, "is_shellcode") else False
+    args.shellcode_entry_point = args.shellcode_entry_point if hasattr(args, "shellcode_entry_point") else None
+    args.shellcode_base = args.shellcode_base if hasattr(args, "shellcode_base") else None
```
Comment (williballenthin) [inline floss/main.py:630]: i'll address this in a subsequent PR that refactors the output rendering (and main() extraction logic).

File: floss/main.py [hunk 4]
```diff
@@ -1062,153 +599,152 @@ def main(argv=None):
...
+            static_unicode_strings = strings.extract_unicode_strings(file_buf, args.min_length)
             result_document.strings.static_strings = list(chain(static_ascii_strings, static_unicode_strings))
             del file_buf
 
-        if options.no_decoded_strings and options.no_stack_strings and not options.should_show_metainfo:
-            if options.json_output_file:
-                create_json_output(result_document, options.json_output_file)
+        if args.no_decoded_strings and args.no_stack_strings and not args.should_show_metainfo:
+            if args.json:
+                print(floss.render.json.render(result_document))
```
Comment (williballenthin) [inline floss/main.py:660]: these checks here are silly and will be refactored in a subsequent PR

**Comment by williballenthin** [comment]
incremental PR that refactors cli arguments and output handling somewhat.

**Review by mr-tz — APPROVED** [review]
Looks good.
My only concern would be keeping all these export scripts around with us not really using or testing them. Does a unified json output provide enough portability so we can maybe avoid it? We can also consider this for the future and decide later.

**Comment by williballenthin** [comment]
yeah, i'm also concerned about these integrations. actually, this motivates moving them out of `main.py`. i actually know how well any of them work these days. so, at the very least, we should be clear about our level of maintenance is for these scripts. i'd hope we can rely on community support, but i'm not sure.

**Pull request merged**